components/raftstore/src/store/peer.rs

```diff
@@ -4343,6 +4515,7 @@ where
         }
 
         if !req.has_admin_request() {
+            ctx.insert(ProposalContext::NO_ADMIN);
```

Comment:
I don't think it's worthy just for an uncommon role to add a flag for all normal requests. You can check for encoding instead. Technically, if it has admin requests, it must contain admin tag.

Reply:
It's costly. I must walk through the previous bytes to check the admin tag, which is close to the overhead of decoding. 
![image](https://user-images.githubusercontent.com/13497871/192224983-fdfcb64f-6f08-433d-8e56-75dd022d6672.png)

Reply:
That's not true. Generally, it only needs to decode 3 varints. The first varint is header field, the second varint is the header length. The third varint is request tag. It can return if the third tag is requests, admin_request or status_request.

Reply:
I see, the payload size can be gotten from the byte right after the tag.

Reply:
done